File: 프로그래머스/l2_H-index.py
def solution(citations):
    answer = 0

    citations.sort(reverse=True)
    h = len(citations)
    now = 0
    while True:
        cnt = now

        for i in range(now, len(citations)):
            if citations[i] >=h:
                cnt+=1
            else:
                now = i-1
                break
        if cnt >=h:
            return h
        h-=1
    
    return answer


solution([3,0,6,1,5])









# def solution(citations):
#     citations.sort(reverse=True)
#     h = len(citations)
#     now = 0
    
#     while True:
#         cnt = now
#         for i in range(now, len(citations)):
#             if citations[i]>=h: cnt +=1
#             else:
#                 now = i-1
#                 break
#         if cnt >=h: return h
#         h -=1

File: 프로그래머스/JAVA/l2_Hindex.java
import java.util.Arrays;

public class l2_Hindex {
    public static void main(String[] args){
        int[] citations = {3,0,6,1,5};
        int answer = solution(citations);
    }

    public static int solution(int[] citations){
        // Array 내림차순에 너무 집착
        // 파이썬 스타일대로 굳이 할 필요 없다.
        Arrays.sort(citations);
        int h = citations.length;
        while (h>=0){
            int cnt = 0;
            for (int i = citations.length-1; i>=0; i--){
                if (citations[i]>=h)
                    cnt ++;
                else
                    break;
            }

            System.out.println(h+" "+cnt);
            if (cnt>=h)
                break;
            h--;
        }
        return h;
    }
}
